### PR TITLE
[WIP] Fix base classes

### PR DESCRIPTION
**Pull request by shrinath-suresh**
This PR contains the updated base component and base executor classes

### CODE REVIEW

File: pytorch_pipeline/components/trainer/component.py [hunk 1]
```diff
@@ -61,13 +80,11 @@ def __init__(
...
                 raise ValueError(f"Unable to load data_module_file - {data_module_file}")
 
-            self.ptl_trainer = Executor().Do(
-                model_class=model_class,
-                data_module_class=data_module_class,
-                data_module_args=data_module_args,
-                module_file_args=module_file_args,
-                trainer_args=trainer_args,
-            )
+            # self.ptl_trainer = Executor().Do(
```
Comment (shrinath-suresh) [inline pytorch_pipeline/components/trainer/component.py]: working on executor restructuring. will enable this once the restructuring is done.

File: pytorch_pipeline/components/trainer/component.py [hunk 2]
```diff
@@ -26,47 +28,34 @@ def __init__(
         :param module_file_args : The arguments of the model class.
         :param trainer_args : These arguments are specific to the pytorch lightning trainer.
         """
-        if not module_file:
```
Comment (chauhang) [inline pytorch_pipeline/components/trainer/component.py:29]: @shrinath-suresh How is the actual input validations going to happen after the code structure change? In the new logic there is no validation check for the Trainer specific parameters.

File: pytorch_pipeline/components/trainer/component.py [hunk 3]
```diff
@@ -26,47 +28,34 @@ def __init__(
...
-            raise ValueError(f"module_file cannot be {module_file}")
 
-        if not data_module_file:
```
Comment (chauhang) [inline pytorch_pipeline/components/trainer/component.py:32]: @shrinath-suresh Same as above

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see the review comments

**Comment by shrinath-suresh** [comment]
@ankan94 @chauhang Lets continue the changes in the following PR - https://github.com/chauhang/pytorch-pipeline/pull/37 . Closing this PR to avoid confusions.

**Pull request closed without merging**